pkg/util/taints/taints.go

```diff
@@ -205,22 +206,23 @@ func deleteTaints(taintsToRemove []v1.Taint, newTaints *[]v1.Taint) ([]error, bo
 	return allErrs, removed
 }
 
-// addTaints adds the newTaints list to existing ones and updates the newTaints List.
-// TODO: This needs a rewrite to take only the new values instead of appended newTaints list to be consistent.
-func addTaints(oldTaints []v1.Taint, newTaints *[]v1.Taint) bool {
+// addedTaints returns the taints that exists in oldTaints but not in newTaints.
+// It returns true when len(oldTains) != (len(*newTaints) + len(addedTaints)).
+// If there are taints in the newTaints but not the oldTaints, return nil and true.
+func addedTaints(oldTaints []v1.Taint, newTaints []v1.Taint) (addedTaints []v1.Taint, lenChanged bool) {
```

Comment:
it is a bit unusual to have method name matching the return value. Perhaps different names would make things better.

Reply:
what do you think about 
func needAddTaints(oldTaints []v1.Taint, newTaints []v1.Taint) (addedTaints []v1.Taint, lenChanged bool)

Reply:
Or CompareTaints?
Or remain the same?

Reply:
Let's keep the same?